consultation_analyser/consultations/views/questions.py

```diff
@@ -33,3 +35,47 @@ def show(request: HttpRequest, consultation_slug: str, section_slug: str, questi
         "multiple_choice": multiple_choice,
     }
     return render(request, "show_question.html", context)
+
+
+def get_applied_filters(request: HttpRequest):
+    return {
+        "keyword": request.GET.get("keyword", ""),
+        "theme": request.GET.get("theme", "All"),
+        "opinion": request.GET.get("opinion", "All"),
+    }
+
+
+def get_filtered_responses(question: models.Question, applied_filters):
+    queryset = models.Answer.objects.filter(question=question, free_text__icontains=applied_filters["keyword"])
+    if applied_filters["theme"] != "All" and applied_filters["theme"] != "No theme":
+        queryset = queryset.filter(theme=applied_filters["theme"])
+    # TO DO: handle answers with "No theme"
```

Comment:
TBC how "No theme" will be represented - so this will change.

There may also be Answers with blank themes - this will be the case when we've uploaded the data but themes haven't yet been assigned.